### PR TITLE
fix(components): Preserve empty features in explanation_spec

### PR DESCRIPTION
**Pull request by evanseabrook**
**Description of your changes:**
[ModelUploadOp](https://google-cloud-pipeline-components.readthedocs.io/en/google-cloud-pipeline-components-1.0.7/google_cloud_pipeline_components.v1.model.html#google_cloud_pipeline_components.v1.model.ModelUploadOp) allows users to specify `explanation_metadata` so that they may provide a list of features and labels to be used with Explainable AI on model upload. This currently works fine for models built with TensorFlow, but for models built using scikit-learn or xgboost, the list of features provided in the `explanation_metadata` argument are dropped.

Because features and outputs provided in `explanation_metadata` should be empty for scikit-learn/xgboost produced models ([per the API spec](https://cloud.google.com/vertex-ai/docs/explainable-ai/configuring-explanations)), the features in the request are dropped by [json_util.recursive_remove_empty](https://github.com/kubeflow/pipelines/blob/master/components/google-cloud/google_cloud_pipeline_components/container/v1/gcp_launcher/upload_model_remote_runner.py#L58).

It looks like [batch predictions](https://github.com/kubeflow/pipelines/blob/master/components/google-cloud/google_cloud_pipeline_components/container/v1/gcp_launcher/batch_prediction_job_remote_runner.py#L32)  are already taken care of by using `ExplanationMetadata.from_json` to parse + evaluate the explanation spec. 

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by evanseabrook** [comment]
@chensun @sasha-gitg Hi folks, I hate to nudge on this one, any chance I can have your review? Happy to put in the legwork if you see any issues with this contribution. My team is currently using scikit-learn on Vertex AI and are affected by the issue that this fixes.

Thanks!

File: components/google-cloud/google_cloud_pipeline_components/container/v1/gcp_launcher/upload_model_remote_runner.py [hunk 1]
```diff
@@ -60,6 +60,12 @@ def upload_model(
                   executor_input, model_spec))
   }
 
+  # Add explanation_spec details back into the request if metadata is non-empty, as sklearn/xgboost input features can be empty.
+  if (('explanation_spec' in model_spec) and
```
Comment (IronPan) [inline components/google-cloud/google_cloud_pipeline_components/container/v1/gcp_launcher/upload_model_remote_runner.py:64]: remove trailing whitespace

File: components/google-cloud/google_cloud_pipeline_components/container/v1/gcp_launcher/upload_model_remote_runner.py [hunk 2]
```diff
@@ -60,6 +60,12 @@ def upload_model(
                   executor_input, model_spec))
   }
 
+  # Add explanation_spec details back into the request if metadata is non-empty, as sklearn/xgboost input features can be empty.
+  if (('explanation_spec' in model_spec) and 
+      ('metadata' in model_spec['explanation_spec']) and
```
Comment (IronPan) [inline components/google-cloud/google_cloud_pipeline_components/container/v1/gcp_launcher/upload_model_remote_runner.py:65]: same here remove trailing whitespace

Comment (evanseabrook) [reply components/google-cloud/google_cloud_pipeline_components/container/v1/gcp_launcher/upload_model_remote_runner.py]: Fixed in both spots; thanks!
